City-of-Helsinki/drupal-module-helfi-tpr
# UHF-4530: Vare ontology words

# Import daycare related ontology words [UHF-4530](https://helsinkisolutionoffice.atlassian.net/browse/UHF-4530)

## What was done
* Renamed OntologyWordDetails entity's `school_details` field to more general `detail_items`. The old data is deleted, so the `tpr_ontology_word_details` migration should be run after the deploy.
* Changed the way how the ontology word IDs can be limited: it's now possible to change the list of IDs from the settings.php file.
* Added daycare related ontology word IDs to the default list for limiting the ontology words.
* Also added the school related IDs to the same list.

## How to install
* Make sure your instance is up and running on latest dev branch.
  * Should be tested at least with the Kasko site.
* Update the TPR module:
    * `composer require drupal/helfi_tpr:UHF-4530-vare-ontology`
* Run `make drush-updb drush-cr`

## How to test
* After the install step, run the migration twice (to ensure proper ontology word details are published):
  * `make shell`
  * `drush migrate:import tpr_ontology_word_details` (twice)
* Go to a page containing the upper secondary school search.
  * Check that the school search and filters still work after the field rename.
